app/reflexes/products_reflex.rb

```diff
@@ -46,6 +46,36 @@ def bulk_update
     render_products_form_with_flash
   end
 
+  def delete_product
+    id = current_id_from_element(element)
+    authorize! :delete, Spree::Product
+    product = product_finder(id).find_product
+    authorize! :delete, product
```

Comment:
Yes, I thought of that before as well. However, I believe this way if the `current_user` doesn't have access to products at all, then we can save the execution of  `ProductScopeQuery`.  What do you think about it?

Reply:
If the user doesn't have access to products at all then they would never reach the admin products screen and trigger a query like this. It would only happen if a user gets logged out or tries to trick the system. So I think that it's a very rare case which we don't need to optimise.

Reply:
makes sense, it's updated now.